pkl/external_reader.go

```diff
@@ -0,0 +1,326 @@
+package pkl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+
+	"github.com/apple/pkl-go/pkl/internal"
+	"github.com/apple/pkl-go/pkl/internal/msgapi"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type ExternalReaderRuntime interface {
+	Run() error
+	Close()
+}
```

Comment:
"Runtime" makes me think of language runtimes.

Some name suggestions:

* `ExternalReaderClient`
* `ExternalReaderService`